mlxtend/frequent_patterns/fpcommon.py

```diff
@@ -82,31 +75,31 @@ def generate_itemsets(generator, num_itemsets, colname_map):
 
 
 def valid_input_check(df):
+
+    if f"{type(df)}" == "<class 'pandas.core.frame.SparseDataFrame'>":
+        msg = ("SparseDataFrame support has been deprecated in pandas 1.0,"
+               " and is no longer supported in mlxtend. "
+               " Please"
+               " see the pandas migration guide at"
+               " https://pandas.pydata.org/pandas-docs/"
+               "stable/user_guide/sparse.html#sparse-data-structures"
+               " for supporting sparse data in DataFrames.")
+        raise TypeError(msg)
+
     if df.size == 0:
         return
-    if hasattr(df, "to_coo") or hasattr(df, "sparse"):
+    if hasattr(df, "sparse"):
         if not isinstance(df.columns[0], str) and df.columns[0] != 0:
             raise ValueError('Due to current limitations in Pandas, '
-                             'if the SparseDataFrame has integer column names,'
+                             'if the sparse format has integer column names,'
                              'names, please make sure they either start '
                              'with `0` or cast them as string column names: '
                              '`df.columns = [str(i) for i in df.columns`].')
 
-    # Fast path: if all columns are boolean, there is nothing to check
-    if Version(pandas_version) >= Version("0.24"):
-        all_bools = ((df.dtypes == pd.SparseDtype(bool)) |
-                     (df.dtypes == bool)).all()
-    else:
-        all_bools = (df.dtypes == bool).all()
+    all_bools = (df.dtypes == bool).all()
```

Comment:
You should have kept
```
   all_bools = ((df.dtypes == pd.SparseDtype(bool)) |
                        (df.dtypes == bool)).all()
```

Reply:
not sure why I deleted this! Now, after adding it back, for some reason 

```python
(df.dtypes == pd.SparseDtype(bool))
```

produces a deprecation warning for some reason. 

    DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    res_values = op(left, right)


Can't reproduce this issue outside the code though.

Reply:
Indeed, that's weird; I do not find how to avoid this warning.

Reply:
This seems to work; `is_bool_dtype` had been introduced in pandas 0.19, so there is no need to check version:
```
# Fast path: if all columns are boolean, there is nothing to checks
all_bools = df.dtypes.apply(pd.api.types.is_bool_dtype).all()
```